unitTests/ui/ValueElementTest.ts

```diff
@@ -118,9 +121,44 @@ export function ValueElementTest() {
 
       it('when the click event on the label is triggered, triggers the change event on the checkbox', () => {
         expect(mockCheckbox.manualEventTriggers).toEqual({});
-        $$(mockLabel.element).trigger('click');
+        mockLabel.element.click();
         expect(countValues(mockCheckbox.manualEventTriggers)).toEqual({ change: 1 });
       });
+
+      it('when the click event on the label is triggered, dismisses search results', () => {
+        const { dismissSearchResults } = facet.facetSearch;
+        expect(dismissSearchResults).not.toHaveBeenCalled();
+        mockLabel.element.click();
+        expect(dismissSearchResults).toHaveBeenCalledTimes(1);
+      });
+
+      it('when the click event on the exclude icon is triggered, dismisses search results', () => {
+        const { dismissSearchResults } = facet.facetSearch;
+        expect(dismissSearchResults).not.toHaveBeenCalled();
+        mockExcludeIcon.element.click();
+        expect(dismissSearchResults).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    describe('selected and unexcluded', () => {
+      beforeEach(() => {
+        valueElement.select();
+        bindEvents();
+      });
+
+      it('when the click event on the label is triggered, dismisses search results', () => {
+        const { dismissSearchResults } = facet.facetSearch;
+        expect(dismissSearchResults).not.toHaveBeenCalled();
+        mockLabel.element.click();
+        expect(dismissSearchResults).toHaveBeenCalledTimes(1);
+      });
+
+      it('when the click event on the exclude icon is triggered, dismisses search results', () => {
+        const { dismissSearchResults } = facet.facetSearch;
+        expect(dismissSearchResults).not.toHaveBeenCalled();
+        mockExcludeIcon.element.click();
+        expect(dismissSearchResults).toHaveBeenCalledTimes(1);
+      });
     });
```

Comment:
I feel this `describe` block and the two `it` blocks below are repetitive tests that can be removed. Unless I am missing something, I think the tests on lines 128 and 135 are enough.

Reply:
Normally, they should be enough, but `select` and `exclude` go through different functions, which caused only one of those to properly dismiss search results.